aya/src/sys/bpf.rs

```diff
@@ -397,11 +401,36 @@ pub(crate) fn bpf_link_create(
         }
     };
     attr.link_create.attach_type = attach_type as u32;
-    attr.link_create.flags = flags;
+
     if let Some(btf_id) = btf_id {
         attr.link_create.__bindgen_anon_3.target_btf_id = btf_id;
     }
 
+    attr.link_create.flags = flags;
+
+    // since kernel 6.6
+    if let Some(expected_revision) = expected_revision {
+        attr.link_create.__bindgen_anon_3.tcx.expected_revision = expected_revision;
+    }
+
+    match (relative_fd, relative_id) {
```

Comment:
if only one of these can be provided, why doesn't the API encode that?

Reply:
Fixed